ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteSessionImpl.java

```diff
@@ -0,0 +1,88 @@
+package org.hibernate.ogm.datastore.ignite.impl;
+
+import java.util.List;
+
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.event.spi.EventSource;
+import org.hibernate.internal.CriteriaImpl;
+import org.hibernate.ogm.OgmSessionFactory;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.OgmCriteriaLoader;
+import org.hibernate.ogm.datastore.ignite.loader.criteria.impl.CriteriaCustomQuery;
+import org.hibernate.ogm.exception.NotSupportedException;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionImpl;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.backend.jta.internal.JtaTransactionCoordinatorImpl;
+
+public class IgniteSessionImpl extends OgmSessionImpl {
```

Comment:
Actually the reason is that a future version of Hibernate ORM will merge the generation tree of JP-QL and Criteria queries and generate a common AST. OGM could then get criteria support for free. It is a longer time coming but than expected but the idea of implementing criteria to ditch the implementation later is not super appealing compared to the usefulness of the feature compared to the other feature ideas we have.